src/madness/mra/key.h

```diff
@@ -174,6 +176,32 @@ namespace madness {
             return dist;
         }
 
+        uint64_t
```

Comment:
I think this code is using the BC to determine how to interpret boxes outside the simulation cell: in the periodic case, shift them one closer to the unit cell if outside it. This would seem to assume that you're at most within 1 simulation cell in the periodic case, and 0 otherwise.

If so, please add comments. I would never have understood this if we hadn't talked about this a few hours ago.